rasa/shared/core/events.py

```diff
@@ -1744,6 +1755,12 @@ def as_dict(self) -> Dict[Text, Any]:
         d["event"] = ActiveLoop.type_name
         return d
 
+    def fingerprint(self) -> Text:
```

Comment:
Had to implement separate fingerprint method because the `as_dict` methods for `LegacyForm` and `LegacyFormValidation` update their `type_name` to be identical to their parent class `type_name` and therefore could run into a problem where 2 different event types have the same fingerprint.

Reply:
Can you put a comment explaining this?